In [61]:
import json
import requests
from time import sleep

In [40]:
url = "https://api.mercadolibre.com/sites/MLB/search?q=chromecast&limit=50"
headers = {"Authorization": "Bearer"}

response = requests.get(url, headers=headers)

response = response.json()

In [41]:
response['paging']

{'total': 494, 'primary_results': 494, 'offset': 0, 'limit': 50}

This means that there are only 494 results for chromecast, and I need 500

In [55]:
url = "https://api.mercadolibre.com/sites/MLB/search?q=switch&limit=50&offset=0"
headers = {"Authorization": "Bearer"}

response = requests.get(url, headers=headers)

response = response.json()

In [56]:
response['paging']

{'total': 110780, 'primary_results': 1000, 'offset': 0, 'limit': 50}

Great

In [62]:
for offset in range(10):
    url = f"https://api.mercadolibre.com/sites/MLB/search?q=switch&limit=50&offset={offset*50}"
    response = requests.get(url, headers=headers)
    if response.status_code == 200:
        with open(f"response_data_{offset}.json", "w") as json_file:
            json.dump(response.json(), json_file)
        sleep(10)
    else:
        print(f"Error {response.status_code}, at offset {offset}, {response}")
        break

No codigo acima estou verificando erros ao verificar o status code, estou cuidando da paginacao ao incorporar o offset e estou limitando o numero de chamadas usando sleep.